lib/rouster.rb

```diff
@@ -640,10 +640,13 @@ def restart(wait=nil)
 
     if wait
       inc = wait.to_i / 10
-      0..wait.each do |e|
+      0.upto(10) do |e|
         @logger.debug(sprintf('waiting for reboot: round[%s], step[%s], total[%s]', e, inc, wait))
-        return true if self.is_available_via_ssh?()
-        sleep inc
+       begin
+         return true if self.is_available_via_ssh?()
+       rescue => ex
+       end
```

Comment:
rather than an empty rescue here, i'd prefer to make a change in `Rouster#is_available_via_ssh?` such that we catch the connection refused error in a general way